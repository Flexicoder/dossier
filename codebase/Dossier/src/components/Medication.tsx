import React, { Component } from 'react';
import { TextInput, View } from 'react-native';
import  styles  from '../assets/styles'

interface MedicationProps {
    drugName?: string | null;
    dose?: string | null;
    frequency?: string | null;
}

export default class Medication extends Component<MedicationProps> {
    constructor(props: MedicationProps) {
        super(props)
        this.state = { drugName: "", dose: "", frequency: "" }
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Drug Name"
                            onChangeText={text => this.setState({ drugName: text})}
                            defaultValue={this.props.drugName} />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                    <TextInput
                        placeholder="Dose"
                        onChangeText={text => this.setState({ dose: text})}
                        defaultValue={this.props.dose} />
                    </View>
                    <View style={styles.textBox}>
                    <TextInput
                        placeholder="Frequency"
                        onChangeText={text => this.setState( { frequency: text })}
                        defaultValue={this.props.frequency}
                        />
                    </View>
                </View>
            </View>
        );
    }
}