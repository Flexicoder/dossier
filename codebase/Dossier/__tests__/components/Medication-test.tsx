
import 'react-native';
import React from 'react';
import Medication from '../../src/components/Medication';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("Medication", () => {
    it('renders correctly', () => {
        const medication = renderer.create(<Medication />);
        expect(medication.toJSON()).toMatchSnapshot();
      });
})


describe("Medication", () => {
    it('contains supplied data', () => {
        const medication = renderer.create(<Medication drugName="Asprin" dose="10mg" frequency="daily" />);
        const stringVersion = JSON.stringify(medication.toJSON())
        expect(stringVersion).toMatch(/Asprin/);
        expect(stringVersion).toMatch(/10mg/);
        expect(stringVersion).toMatch(/daily/);
    });
})