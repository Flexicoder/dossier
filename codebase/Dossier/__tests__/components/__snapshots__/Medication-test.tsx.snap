// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Medication renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Drug Name"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Dose"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Frequency"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
